src/Components/Auction/PostRegistrationModal.tsx

```diff
@@ -0,0 +1,111 @@
+import { Box, Button, CheckCircleIcon, Modal, Serif } from "@artsy/palette"
+import React, { useEffect, useState } from "react"
+
+export type ContentKey =
+  | "registrationConfirmed"
+  | "registrationPending"
+  | "bidPending"
+
+interface Props {
+  onClose: () => void
+  contentKey: ContentKey
+}
+
+type ModalContent = React.ComponentType<{ onClick: () => void }>
+
+const BidPending: ModalContent = ({ onClick }) => {
+  return (
+    <>
+      <RegistrationPendingHeader />
+      <Serif my={3} size="3t">
+        We're sorry, your bid could not be placed.
+      </Serif>
+      <ReviewingRegistrationContent />
+      <ViewWorksButton onClick={onClick} />
+    </>
+  )
+}
+
+const RegistrationPending: ModalContent = ({ onClick }) => {
+  return (
+    <>
+      <RegistrationPendingHeader />
+      <ReviewingRegistrationContent />
+      <ViewWorksButton onClick={onClick} />
+    </>
+  )
+}
+const RegistrationComplete: ModalContent = ({ onClick }) => {
+  return (
+    <>
+      <Serif size="6">Registration complete</Serif>
+      <CheckCircleIcon mt={2} height="28px" width="28px" fill="green100" />
+      <Serif mt={2} mb={3} size="3t">
+        Thank you for registering.
+        <br />
+        You’re now eligible to bid on lots in this sale.
+      </Serif>
+      <Button width="100%" onClick={onClick}>
+        Start bidding
+      </Button>
+    </>
+  )
+}
+
+const contentFor: { [key in ContentKey]: ModalContent } = {
+  registrationConfirmed: RegistrationComplete,
+  registrationPending: RegistrationPending,
+  bidPending: BidPending,
+}
```

Comment:
Unfortunately this little map demanded that all the content components be defined above it, which was annoying. I considered restructuring some files to be able to import them from elsewhere and avoid the ugliness but didn't go that far yet.